### PR TITLE
Alerting: Changes to evaluation group step

### LINKED ISSUE #68097: Evaluation group selection improvement
Screenshot 2023-05-10 at 12 30 08

**New evaluation group selection modal**

Be able to select folder and group in one go:

Screenshot 2023-05-10 at 12 29 24

### PR DESCRIPTION
**Pull request by VikaCep**
**What is this feature?**

Improves the evaluation group step in the alert rule form.

**Why do we need this feature?**

To make this step easier to understand.

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/68097

Prototype: https://www.figma.com/proto/AYMboJsnYgSAAdYM9hoMLn/Alerting?page-id=0%3A1&type=design&node-id=1323-30604&viewport=-1512%2C-6635%2C0.65&t=21RvsHATbNrd64MU-1&scaling=min-zoom&starting-point-node-id=1322%3A29730&mode=design

UI changes from rule form
image

Creating folder modal
image

Creating evaluation group modal
image

Editing evaluation group modal
image

Screen recording creating a folder

Screen recording creating an evaluation group

### CODE REVIEW

**Comment by gillesdemey** [comment]
I think we might want to disable the "New evaluation group" when you haven't selected a folder yet since it doesn't make any sense to create one (yet) :)

File: public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx [hunk 1]
```diff
@@ -164,50 +116,36 @@ function FolderGroupAndEvaluationInterval({
 
   return (
     <div>
-      <FolderAndGroup />
+      <FolderAndGroup groupfoldersForGrafana={groupfoldersForGrafana} />
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx:119]: It seems a bit odd to pass `AsyncRequestState` around like this, it seems like we only access `result` in the child components. Is it possible to refactor this so we only pass the results in `<FolderAndGroup />`?

**Review by gillesdemey — APPROVED** [review]
small nit, code and functionality are 👌

File: public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx (hunk 1 above)
Comment (VikaCep) [reply public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx:119]: Done!

**Pull request merged**